src/webgl/shaders/lighting.glsl

```diff
@@ -119,8 +119,8 @@ vec3 calculateImageDiffuse( vec3 vNormal, vec3 vViewPosition ){
 
 vec3 calculateImageSpecular( vec3 vNormal, vec3 vViewPosition ){
   vec3 worldCameraPosition =  vec3(0.0, 0.0, 0.0);
-  vec3 worldNormal = normalize(vNormal);
-  vec3 lightDirection = normalize( vViewPosition - worldCameraPosition );
+  vec3 worldNormal = normalize(vNormal * uCameraMatrix);
+  vec3 lightDirection = normalize( vViewPosition * uCameraMatrix - worldCameraPosition );
```

Comment:
The type of transform you do for normals and for positions are generally different (normals must always be perpendicular to the surface, while positions can squish; see [the diagram in this SO answer](https://stackoverflow.com/a/26477624) for an idea of why we need something different for normals.) So generally you won't find yourself multiplying both positions and normals by the same matrix.

In this case, I think we only need to do it to the normal, and probably only after doing the reflection:
```js
vec3 worldNormal = normalize(vNormal);
vec3 lightDirection = normalize( vViewPosition - worldCameraPosition );
vec3 R = reflect(lightDirection, worldNormal) * uCameraMatrix;
```

The idea is that, without modifying the normal or position here, we would be looking up the texture correctly if the sphere map that surrounds the scene is un-rotated. To account for its rotation, only at the end would we apply the rotation matrix to it.